editor/project_manager.cpp

```diff
@@ -1422,7 +1475,28 @@ void ProjectManager::_open_project() {
 		multi_open_ask->set_text(TTR("Are you sure to open more than one project?"));
 		multi_open_ask->popup_centered_minsize();
 	} else {
-		_open_project_confirm();
+
+		// Update the project settings or don't open
+		String path = EditorSettings::get_singleton()->get("projects/" + selected_list.front()->key());
+		String conf = path + "/project.godot";
+
+		ProjectSettings *current = memnew(ProjectSettings);
+		if (current->setup(path, "") != OK) {
+			dialog_error->set_text(TTR("Can't open project"));
+			dialog_error->popup_centered_minsize();
+			return;
+		}
+
+		if (!current->is_settings_text_up_to_date_custom(conf)) {
+			confirm_update_settings_args.project_settings = current;
+			confirm_update_settings_args.path = conf;
+			ask_update_settings->set_text(vformat(TTR("Do you want to update the project settings file : %s ?\nWarning : you will not be able to open the project with previous versions of Godot anymore."), conf));
```

Comment:
IMO we shouldn't tell users about a project settings file specifically, we should just mention Godot versions. Something more like "The project %s was opened in a previous version of Godot, would you like to upgrade it to Godot [version]? Warning: You will not be able to open the project with previous versions of Godot anymore."

Reply:
I don't know, I think it's helpful to know which file is going to be modified. @akien-mga WDYT ?